# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/debugger/Script.cpp
# Commit: e802face8e04
# Full Hash: e802face8e042d03933c1495d36fd02a9def7dd9
# Author: Brian Hackett <bhackett1024@gmail.com>
# Date: 2019-07-25 21:51:57
# Regressor Bug: 1554524
# File Overlap Count: 1
# Description:
#   Bug 1554524 Part 4 - Add Debugger interface for instrumenting scripts, r=jimb,tcampbell.
# ==============================================================================

diff -r fc204daba6f3 -r e802face8e04 js/src/debugger/Script.cpp
--- a/js/src/debugger/Script.cpp	Mon Jul 08 09:39:52 2019 -1000
+++ b/js/src/debugger/Script.cpp	Mon Jul 08 09:41:21 2019 -1000
@@ -2084,6 +2084,28 @@
 }
 
 /* static */
+bool DebuggerScript::setInstrumentationId(JSContext* cx, unsigned argc,
+                                          Value* vp) {
+  THIS_DEBUGSCRIPT_SCRIPT_MAYBE_LAZY(cx, argc, vp, "setInstrumentationId", args,
+                                     obj);
+
+  if (!obj->getInstrumentationId().isUndefined()) {
+    JS_ReportErrorASCII(cx, "Script instrumentation ID is already set");
+    return false;
+  }
+
+  if (!args.get(0).isNumber()) {
+    JS_ReportErrorASCII(cx, "Script instrumentation ID must be a number");
+    return false;
+  }
+
+  obj->setReservedSlot(INSTRUMENTATION_ID_SLOT, args.get(0));
+
+  args.rval().setUndefined();
+  return true;
+}
+
+/* static */
 bool DebuggerScript::construct(JSContext* cx, unsigned argc, Value* vp) {
   JS_ReportErrorNumberASCII(cx, GetErrorMessage, nullptr, JSMSG_NO_CONSTRUCTOR,
                             "Debugger.Script");
@@ -2119,6 +2141,7 @@
     JS_FN("getOffsetsCoverage", getOffsetsCoverage, 0, 0),
     JS_FN("getSuccessorOffsets", getSuccessorOffsets, 1, 0),
     JS_FN("getPredecessorOffsets", getPredecessorOffsets, 1, 0),
+    JS_FN("setInstrumentationId", setInstrumentationId, 1, 0),
 
     // The following APIs are deprecated due to their reliance on the
     // under-defined 'entrypoint' concept. Make use of getPossibleBreakpoints,